e2e/tests/core/02-client/client_test.go

```diff
@@ -71,7 +73,66 @@ func (s *ClientTestSuite) QueryAllowedClients(ctx context.Context, chain ibc.Cha
 	return res.Params.AllowedClients
 }
 
-func (s *ClientTestSuite) TestClientUpdateProposal_Succeeds() {
+// TestScheduleIBCUpgrade_Succeeds tests that a governance proposal to schedule an IBC software upgrade is successful.
+func (s *ClientTestSuite) TestScheduleIBCUpgrade_Succeeds() {
+	t := s.T()
+	ctx := context.TODO()
+
+	_, _ = s.SetupChainsRelayerAndChannel(ctx)
+	chainA, chainB := s.GetChains()
+	chainAWallet := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+
+	const planHeight = int64(75)
+	var newChainID string
+
+	t.Run("execute proposal for MsgIBCSoftwareUpgrade", func(t *testing.T) {
+		authority, err := s.QueryModuleAccountAddress(ctx, govtypes.ModuleName, chainA)
+		s.Require().NoError(err)
+		s.Require().NotNil(authority)
+
+		clientState, err := s.QueryClientState(ctx, chainB, ibctesting.FirstClientID)
+		s.Require().NoError(err)
+
+		originalChainID := clientState.(*ibctm.ClientState).ChainId
+		revisionNumber := clienttypes.ParseChainID(fmt.Sprintf("%s-%d", originalChainID, 1))
+		// increment revision number even with new chain ID to prevent loss of misbehaviour detection support
+		newChainID, err = clienttypes.SetRevisionNumber(fmt.Sprintf("%s-%d", originalChainID, 1), revisionNumber+1)
+		s.Require().NoError(err)
+		s.Require().NotEqual(originalChainID, newChainID)
+
+		upgradedClientState := clientState.(*ibctm.ClientState)
+		upgradedClientState.ChainId = newChainID
+
+		scheduleUpgradeMsg, err := clienttypes.NewMsgIBCSoftwareUpgrade(
+			authority.String(),
+			types.Plan{
+				Name:   "upgrade-client",
+				Height: planHeight,
+			},
+			upgradedClientState,
+		)
+		s.Require().NoError(err)
+		s.ExecuteGovV1Proposal(ctx, scheduleUpgradeMsg, chainA, chainAWallet, 1)
+	})
+
+	t.Run("check that IBC software upgrade has been scheduled successfully on chainA", func(t *testing.T) {
+		// checks there is an upgraded client state stored
+		cs, err := s.QueryUpgradedClientState(ctx, chainA, ibctesting.FirstClientID)
+		s.Require().NoError(err)
+
+		upgradedClientState := cs.(*ibctm.ClientState)
```

Comment:
[nit] might be worth doing an assertion on `ok` from the cast. Not that important since the next line would fail anyway, but the error might be misleading if that was the case.